packages/block-library/src/block/index.php

```diff
@@ -87,6 +87,26 @@ function render_block_core_block( $attributes ) {
 		add_filter( 'render_block_context', $filter_block_context, 1 );
 	}
 
+	$ignored_hooked_blocks = get_post_meta( $attributes['ref'], '_wp_ignored_hooked_blocks', true );
```

Comment:
I couldn't find a better place, so I will ask here. How feasible would it be to move the logic to `apply_block_hooks_to_content` in WordPress core? A very similar logic exists now in a few places. As the helper function gets the current post object as param, it should be able to handle everything as it happens inside the `rest_pre_insert_{$post_type}` filter.

Reply:
Hah, I just had a similar thought yesterday!

I'll need to check if we can really absorb this logic into `apply_block_hooks_to_content`. Note that templates and template parts are still handled quite differently from posts, synced patterns, and navigation blocks:

1. They're instances of the `WP_Block_Template` class rather than `WP_Post`.
2. While first/last child insertion [is implemented](https://github.com/WordPress/wordpress-develop/pull/6867) for template parts, it is _not_ implemented for templates. (The reason is that there is a Template Part _block_, `core/template-part`, that can be used as an anchor block; but of course, there's no Template block, so it cannot be used as an anchor block for Block Hooks.)
3. I need to think a bit more about patterns.

Anyway, I'll file a ticket for this and will look into it in January 😊

Reply:
> Anyway, I'll file a ticket for this and will look into it in January 😊

https://core.trac.wordpress.org/ticket/62716